File: simulation/backend/models/warehouse/logistics.py
"""Logistics and task models."""
from __future__ import annotations

from typing import Literal
from pydantic import BaseModel


TaskType = Literal["inbound", "outbound", "transfer", "replenishment"]
TaskStatus = Literal["pending", "in_progress", "completed", "cancelled"]


class TaskItem(BaseModel):
    item_code: str
    item_name: str
    qty: float
    uom: str = "PCS"


class LogisticsTask(BaseModel):
    ref: str
    type: TaskType
    status: TaskStatus
    priority: int = 3
    source_dock: str | None = None
    target_dock: str | None = None
    items: list[TaskItem] = []
    assigned_vehicle: str | None = None
    eta: float | None = None
    completed_at: float | None = None
    created_at: float


class LogisticsStats(BaseModel):
    total_inbound: int = 0
    total_outbound: int = 0
    avg_processing_time: float = 0
    dock_utilization: float = 0


class DockDetail(BaseModel):
    ref: str
    direction: Literal["inbound", "outbound"]
    name: str
    x: float
    z: float
    slots: list[DockSlot]
    utilization: float = 0


class DockSlot(BaseModel):
    ref: str
    status: Literal["available", "occupied", "scheduled"] = "available"
    task: str | None = None
    vehicle: str | None = None


class FloorFull(BaseModel):
    shell: ShellBlueprint | None = None
    zones: list[Zone] = []
    facilities: list[Facility] = []
    docks: list[Dock] = []


# Forward references for FloorFull
from backend.models.warehouse.zone import (
    ShellBlueprint,
    Zone,
    Facility,
    Dock,
)

File: rcs/frontend/src/views/simulation/warehouse/types/logistics.ts
import type { Zone, Facility, Dock, ShellBlueprint } from './zone'

export interface FloorFull {
  shell?: ShellBlueprint
  zones?: Zone[]
  facilities?: Facility[]
  docks?: Dock[]
}

export type TaskType = 'inbound' | 'outbound' | 'transfer' | 'replenishment'
export type TaskStatus = 'pending' | 'in_progress' | 'completed' | 'cancelled'

export interface TaskItem {
  item_code: string
  item_name: string
  qty: number
  uom: string
}

export interface LogisticsTask {
  ref: string
  type: TaskType
  status: TaskStatus
  priority: number
  source_dock?: string
  target_dock?: string
  items: TaskItem[]
  assigned_vehicle?: string
  eta?: number
  completed_at?: number
  created_at: number
}

export interface LogisticsStats {
  total_inbound: number
  total_outbound: number
  avg_processing_time: number
  dock_utilization: number
}

export interface DockDetail {
  ref: string
  direction: 'inbound' | 'outbound'
  name: string
  x: number
  z: number
  slots: DockSlot[]
  utilization: number
}

export interface DockSlot {
  ref: string
  status: 'available' | 'occupied' | 'scheduled'
  task?: string
  vehicle?: string
}
